apps/webservice/src/app/api/github/webhook/workflow/handler.ts

```diff
@@ -21,31 +21,54 @@ const convertStatus = (
 ): schema.JobStatus =>
   status === JobStatus.Completed ? JobStatus.Completed : JobStatus.InProgress;
 
+const extractUuid = (str: string) => {
+  const uuidRegex =
+    /\b[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\b/;
+  const match = uuidRegex.exec(str);
+  return match ? match[0] : null;
+};
+
+const getJob = async (externalId: number, name: string) => {
+  const jobFromExternalId = await db
+    .select()
+    .from(schema.job)
+    .where(eq(schema.job.externalId, externalId.toString()))
+    .then(takeFirstOrNull);
+
+  if (jobFromExternalId != null) return jobFromExternalId;
+
+  const uuid = extractUuid(name);
+  if (uuid == null) return null;
+
+  return db
+    .select()
+    .from(schema.job)
+    .where(eq(schema.job.id, uuid))
+    .then(takeFirstOrNull);
+};
+
 export const handleWorkflowWebhookEvent = async (event: WorkflowRunEvent) => {
   const {
     id,
     status: externalStatus,
     conclusion,
     repository,
+    name,
   } = event.workflow_run;
 
+  const job = await getJob(id, name);
+  if (job == null) return;
+
   const status =
     conclusion != null
       ? convertConclusion(conclusion)
       : convertStatus(externalStatus);
 
-  const job = await db
+  const externalId = id.toString();
+  await db
     .update(schema.job)
-    .set({ status })
-    .where(eq(schema.job.externalId, id.toString()))
-    .returning()
-    .then(takeFirstOrNull);
-
-  // Addressing a race condition: When the job is created externally on GitHub,
-  // it triggers a webhook event. However, our system hasn't updated the job with
-  // the externalRunId yet, as it depends on the job's instantiation. Therefore,
-  // the first event lacks the run ID, so we skip it and wait for the next event.
-  if (job == null) return;
+    .set({ status, externalId })
```

Comment:
you can also generate the url at this point? I think

Reply:
we are generating the URL below